docs/recipes/stencil.md

```diff
@@ -0,0 +1,87 @@
+# Usage with Stencil
+
+[Stencil](https://stenciljs.com/) web components work very well with XState.
+
+### `src/helpers/toggle-machine.ts`
+
+```js
+import { createMachine } from "@xstate/fsm"
+
+// this machine is completely decoupled from stencil
+export const toggleMachine = createMachine({
+  id: "toggle",
+  initial: "inactive",
+  states: {
+    inactive: {
+      on: { toggle: "active" }
+    },
+    active: {
+      on: { toggle: "inactive" }
+    }
+  }
+});
+```
+
+### `src/components/toggle/toggle.tsx`
+
+
+Add a `state` property to your component, decorated with `@State` so that it triggers a re-render when changed.
+
+On `componentWillLoad`, interpret the `toggleMachine` and listen for state transitions.
+
+After a transition has occured, the `state` property is set to the machine's new state, triggering a re-render.
+
+```js
+import { Component, h, State, Prop } from "@stencil/core";
+import { interpret } from "xstate";
+import { toggleMachine } from "../helpers/toggle-machine";
+
+
+@Component({
+  tag: "my-toggle",
+  styleUrl: "toggle.css",
+  shadow: true
+})
+export class Toggle {
+  @State() state;

```

Comment:
Is there a type for this? I think it's `StateMachine.Service`? Like

```ts
import { interpret, StateMachine } from '@xstate/fsm'

export class Toggle {
  @State() state: StateMachine.Service;
}
```

?

Reply:
Ah no sry that's for `private _service`... there's also `StateMachine.State` though for this I think.

Reply:
I've got a much more strictly typed version here: https://github.com/edsilv/stencil-xstate-demo/blob/master/src/components/toggle/toggle.tsx
Was trying to keep it simpler for the recipe, but maybe it's better not to?

Reply:
Ah I see yeah those types are quite verbose...